installer_mac/make_installer.sh

```diff
@@ -25,36 +25,55 @@ fi
 
 # locations
 
-VST2="../products/Surge.vst"
-VST3="../products/Surge.vst3"
-AU="../products/Surge.component"
+PRODUCTS="../products/"
+VST2="Surge.vst"
+VST3="Surge.vst3"
+AU="Surge.component"
 RSRCS="../resources/data"
 
 OUTPUT_BASE_FILENAME="Surge-$VERSION-Setup"
 
+build_flavor()
+{
+    TMPDIR=installer/tmp-pkg
+    flavor=$1
+    flavorprod=$2
+    ident=$3
+    loc=$4
+
+    echo --- BUILDING Surge_${flavor}.pkg ---
+    
+    mkdir $TMPDIR
+    mv $PRODUCTS/$flavorprod $TMPDIR
+    
+    
+	pkgbuild --analyze --root $TMPDIR Surge_${flavor}.plist
+    echo "Plist for Package:"
+    cat Surge_${flavor}.plist
+	pkgbuild --root $TMPDIR --component-plist Surge_${flavor}.plist --identifier $ident --version $VERSION --install-location $loc Surge_${flavor}.pkg
+
+	rm Surge_${flavor}.plist
+    mv $TMPDIR/${flavorprod} $PRODUCTS
+    
+    rmdir $TMPDIR
+}
 
 # try to build VST2 package
 
-if [[ -d $VST2 ]]; then
-	pkgbuild --analyze --root "$VST2" Surge_VST2.plist
```

Comment:
If I'm reading [this man page](http://www.manpagez.com/man/1/pkgbuild/) correctly, it looks like maybe we just need to create the `.component`,  `.vst`, and `.vst3` in an empty directory, skip the `--analyze` and just run `pkgbuild` with that previously empty dir that contains the corresponding `Surge.component|vst|vst3`?

Reply:
You need a plist file of some form otherwise it doesn't work. As evidenced by having a wrong plist file before these diffs :)

The requirement to edit that plist file is infrequent and in our case you don't need to which is why we can analyze each time. We could also analyze once and edit but that would be more fragile

Reply:
Creating it in an empty directory is actually what my change does! You are reading the red code not the green code :)

Reply:
But I still run analyze because once it is in an empty directory you can get the plist file which we may want to mod in the future. And which I am pretty sure you need.

Reply:
Yes, I understand the red code is the removed stuff. I really don't think we need the `.plist` at all if that document is right. My point is just change the `.vst` etc. build destination, then we don't need to make temp directories and this extra `--analyze` step. I'd say it's at least worth a try. I won't have time to poke around for a few hours, though...

Reply:
Oh right so make a products/vst/ products/vst3/ products/au and then don't do the copy rigamarole. Modify premake and build-osx.sh also to account for those new locations. That's a bigger diff but I think it might work and avoid that mv / mv-back dance I did.

Before you did that though I would see if it works without the plist file at all. Because if you still need to run analyze anyway then that's a lot of work to save one mv / mv back

Reply:
And my expertise here is low. I'm reading the same man pages you are. So I could easily be wrong about needing that plist. Happy to not do that step if we don't need to.

Reply:
My expertise is also low ;)

But keeping this stuff as simple as possible would help others read it later. I think the diff might actually be smaller; and if we can remove code 🎉 !

Reply:
Yup. If you can make that work then that’s a better diff. But at least we know the problem now and the shape of the solution. Happy to use a diff youbwrite rather than a diff I do!

Reply:
@kzantow you were correct we didn't need to run analyze. But moving the product to a tmp dir seems like the smallest change we could make. What do you think of the version here? Let us both remove code and not change the build system! And it works for me! (@esaruoho is testing now too)